### ISSUE TITLE
New temp-named notebook file is created and left in repo.

### ISSUE THREAD

**Issue by miguelsolorio** [microsoft/vscode-jupyter#7078]
## Environment data

**VS Code**
-   Version: 1.60.0-insider (Universal)
-   Commit: 7c25c174726ab969db3f99058cdf24bcfae1cfbe
-   Date: 2021-08-11T05:13:17.497Z
-   Electron: 13.1.8
-   Chrome: 91.0.4472.164
-   Node.js: 14.16.0
-   V8: 9.1.269.39-electron.0
-   OS: Darwin arm64 20.6.0

**Jupyter**
-   Jupyter Extension version (available under the Extensions sidebar): v2021.8.2025984474
-   Python Extension version (available under the Extensions sidebar): v2021.9.1117361286-dev
-   OS (Windows | Mac | Linux distro) and version: macOS M1
-   Python and/or Anaconda version: Python 3.8.8 64-bit
-   Type of virtual environment used (N/A | venv | virtualenv | conda | ...): conda
-   Jupyter server running: Local

## Expected behaviour

For no new files to be created

## Actual behaviour

Creates a new file that is not usable

## Steps to reproduce:

1. Open a notebook
2. Select kernel
3. Run notebook
4. Switch interpreter
5. :bug: See file get created and remain

I haven't been able to fully repro this but have been seeing this happen within the last week. I think it has to do with switching interpreters/kernels.

Might be related to https://github.com/microsoft/vscode-jupyter/issues/6510 but I wasn't previously seeing this.

![image](https://user-images.githubusercontent.com/35271042/129095191-155d1a37-3a02-4e37-a43e-729dafc869ef.png)

<img width="1024" alt="CleanShot 2021-08-11 at 13 00 52@2x" src="https://user-images.githubusercontent.com/35271042/129095164-83ce6c34-7266-424a-8e5f-babe491a15bd.png">

**Comment by rchiodo** [comment]
That file is created here: https://github.com/microsoft/vscode-jupyter/blob/8e06fe952f76a3029e5b506e863fd628d5036157/src/client/datascience/jupyter/jupyterSession.ts#L206

But then should be deleted here:
https://github.com/microsoft/vscode-jupyter/blob/8e06fe952f76a3029e5b506e863fd628d5036157/src/client/datascience/jupyter/jupyterSession.ts#L294

So somewhere in the middle of that process your kernel might have died? Been switched? You don't happen to have the logs from the output tab for 'Jupyter' do you?

**Comment by hotplot** [comment]
I have a similar issue, except it's much more repeatable. When I use Python Interactive files, the directory that they are saved in ends up littered with many (eventually dozens) of these temp files. 

A possible point of difference between my setup and @misolori's is that I'm working remotely on an HPC cluster using the "Remote - SSH" extension. It does not appear to have anything to do with switching kernels, and I rarely if ever do this once a project is setup. It may be a different bug.

I have attached the relevant part of the Jupyter log below.

## Environment data

- Version: 1.61.2 (user setup)
- Commit: 6cba118ac49a1b88332f312a8f67186f7f3c1643
- Date: 2021-10-19T14:57:20.575Z
- Electron: 13.5.1
- Chrome: 91.0.4472.164
- Node.js: 14.16.0
- V8: 9.1.269.39-electron.0
- OS: Windows_NT x64 10.0.19041

-   Jupyter Extension version (available under the Extensions sidebar): v2021.9.1101343141
-   Python Extension version (available under the Extensions sidebar): v2021.10.1365161279
-   OS (Windows | Mac | Linux distro) and version: **Local OS**: Windows_NT x64 10.0.19041, **Remote OS**: CentOS Linux 7 3.10.0-1160.42.2.el7.x86_64
-   Python and/or Anaconda version: Python 3.6, 3.7, 3.9 exhibit the issue
-   Type of virtual environment used (N/A | venv | virtualenv | conda | ...): conda
-   Jupyter server running: Local

## Expected behaviour

Python interactive cells are run and no temp files are left in the repo

## Actual behaviour

Python interactive cells run correctly, but numerous temp files are left in the same directory as the interactive file

## Steps to reproduce:

1. Create a `*.py` file and add an interactive cell with `# %%`
2. Execute the interactive cell

## Logs

```
[I 13:31:18.254 NotebookApp] Kernel started: 340c1493-7f52-4f0a-92b0-e80c0c58ce7e, name: python3613jvsc74a57bd0865768383b38e3868f2c611ccdf592e4ddd497867759542f2788f434be7540ef
Info 2021-10-22 13:31:18: Waiting for idle on (kernel): 340c1493-7f52-4f0a-92b0-e80c0c58ce7e -> unknown
Info 2021-10-22 13:31:18: Python Daemon (pid: 18855): write to stderr: [W 13:31:18.263 NotebookApp] delete /notebooks/rename.py-50983818-1b6e-414f-bb20-1bdd210cbea8.ipynb

[W 13:31:18.263 NotebookApp] delete /notebooks/rename.py-50983818-1b6e-414f-bb20-1bdd210cbea8.ipynb
Info 2021-10-22 13:31:18: Python Daemon (pid: 18855): write to stderr: [E 13:31:18.267 NotebookApp] Uncaught exception DELETE /api/contents/notebooks/rename.py-50983818-1b6e-414f-bb20-1bdd210cbea8.ipynb?1634862678257 (127.0.0.1)
    HTTPServerRequest(protocol='http', host='localhost:8888', method='DELETE', uri='/api/contents/notebooks/rename.py-50983818-1b6e-414f-bb20-1bdd210cbea8.ipynb?1634862678257', version='HTTP/1.1', remote_ip='127.0.0.1')
    Traceback (most recent call last):
      File "/pypath/lib/python3.9/site-packages/tornado/web.py", line 1704, in _execute
        result = await result
      File "/pypath/lib/python3.9/site-packages/tornado/gen.py", line 234, in wrapper
        yielded = ctx_run(next, result)
      File "/pypath/lib/python3.9/site-packages/notebook/services/contents/handlers.py", line 237, in delete
        yield maybe_future(cm.delete(path))
      File "/pypath/lib/python3.9/site-packages/notebook/services/contents/manager.py", line 279, in delete
        self.delete_file(path)
      File "/pypath/lib/python3.9/site-packages/notebook/services/contents/filemanager.py", line 533, in delete_file
        send2trash(os_path)
      File "/pypath/lib/python3.9/site-packages/send2trash/plat_other.py", line 211, in send2trash
        trash_move(path_b, dest_trash, topdir)
      File "/pypath/lib/python3.9/site-packages/send2trash/plat_other.py", line 117, in trash_move
        os.rename(src, op.join(filespath, destname))
    OSError: [Errno 18] Invalid cross-device link: b'/workspace/hrrsxb/AIP/projname/notebooks/rename.py-50983818-1b6e-414f-bb20-1bdd210cbea8.ipynb' -> b'/home/hrrsxb/.local/share/Trash/files/rename.py-50983818-1b6e-414f-bb20-1bdd210cbea8.ipynb'

[E 13:31:18.267 NotebookApp] Uncaught exception DELETE /api/contents/notebooks/rename.py-50983818-1b6e-414f-bb20-1bdd210cbea8.ipynb?1634862678257 (127.0.0.1)
    HTTPServerRequest(protocol='http', host='localhost:8888', method='DELETE', uri='/api/contents/notebooks/rename.py-50983818-1b6e-414f-bb20-1bdd210cbea8.ipynb?1634862678257', version='HTTP/1.1', remote_ip='127.0.0.1')
    Traceback (most recent call last):
      File "/pypath/lib/python3.9/site-packages/tornado/web.py", line 1704, in _execute
        result = await result
      File "/pypath/lib/python3.9/site-packages/tornado/gen.py", line 234, in wrapper
        yielded = ctx_run(next, result)
      File "/pypath/lib/python3.9/site-packages/notebook/services/contents/handlers.py", line 237, in delete
        yield maybe_future(cm.delete(path))
      File "/pypath/lib/python3.9/site-packages/notebook/services/contents/manager.py", line 279, in delete
        self.delete_file(path)
      File "/pypath/lib/python3.9/site-packages/notebook/services/contents/filemanager.py", line 533, in delete_file
        send2trash(os_path)
      File "/pypath/lib/python3.9/site-packages/send2trash/plat_other.py", line 211, in send2trash
        trash_move(path_b, dest_trash, topdir)
      File "/pypath/lib/python3.9/site-packages/send2trash/plat_other.py", line 117, in trash_move
        os.rename(src, op.join(filespath, destname))
    OSError: [Errno 18] Invalid cross-device link: b'/workspace/hrrsxb/AIP/projname/notebooks/rename.py-50983818-1b6e-414f-bb20-1bdd210cbea8.ipynb' -> b'/home/hrrsxb/.local/share/Trash/files/rename.py-50983818-1b6e-414f-bb20-1bdd210cbea8.ipynb'
Info 2021-10-22 13:31:18: Python Daemon (pid: 18855): write to stderr: [W 13:31:18.267 NotebookApp] Unhandled error

[W 13:31:18.267 NotebookApp] Unhandled error
Info 2021-10-22 13:31:18: Python Daemon (pid: 18855): write to stderr: [E 13:31:18.268 NotebookApp] {
      "Authorization": "token 5d6347922646d0c5ff0003f86d55710d67c27a47dee426bb",
      "Content-Type": "application/json",
      "Accept": "*/*",
      "User-Agent": "node-fetch/1.0 (+https://github.com/bitinn/node-fetch)",
      "Accept-Encoding": "gzip,deflate",
      "Connection": "close",
      "Host": "localhost:8888"
    }

[E 13:31:18.268 NotebookApp] {
      "Authorization": "token 5d6347922646d0c5ff0003f86d55710d67c27a47dee426bb",
      "Content-Type": "application/json",
      "Accept": "*/*",
      "User-Agent": "node-fetch/1.0 (+https://github.com/bitinn/node-fetch)",
      "Accept-Encoding": "gzip,deflate",
      "Connection": "close",
      "Host": "localhost:8888"
    }
Info 2021-10-22 13:31:18: Python Daemon (pid: 18855): write to stderr: [E 13:31:18.268 NotebookApp] 500 DELETE /api/contents/notebooks/rename.py-50983818-1b6e-414f-bb20-1bdd210cbea8.ipynb?1634862678257 (127.0.0.1) 6.190000ms referer=None

[E 13:31:18.268 NotebookApp] 500 DELETE /api/contents/notebooks/rename.py-50983818-1b6e-414f-bb20-1bdd210cbea8.ipynb?1634862678257 (127.0.0.1) 6.190000ms referer=None
Info 2021-10-22 13:31:18: Finished waiting for idle on (kernel): 340c1493-7f52-4f0a-92b0-e80c0c58ce7e -> idle
Info 2021-10-22 13:31:18: Started session 5bc6e4e2-0c49-4982-b181-2202a81f877a
Info 2021-10-22 13:31:18: Waiting for idle (session) 5bc6e4e2-0c49-4982-b181-2202a81f877a
Info 2021-10-22 13:31:18: Waiting for idle on (kernel): 340c1493-7f52-4f0a-92b0-e80c0c58ce7e -> idle
Info 2021-10-22 13:31:18: Finished waiting for idle on (kernel): 340c1493-7f52-4f0a-92b0-e80c0c58ce7e -> idle
Info 2021-10-22 13:31:18: Finished connecting 5bc6e4e2-0c49-4982-b181-2202a81f877a
Info 2021-10-22 13:31:18: UpdateWorkingDirectoryAndPath in Kernel
Info 2021-10-22 13:31:18: IPyWidgetScriptSource.initialize
Info 2021-10-22 13:31:18: Registering commtarget jupyter.widget
Info 2021-10-22 13:31:18: IPyWidgetMessageDispatcher.initialize
Info 2021-10-22 13:31:18: IPyWidgetMessageDispatcher.initialize
Info 2021-10-22 13:31:18: IPyWidgetMessageDispatcher.initialize
Info 2021-10-22 13:31:19: changeDirectoryIfPossible
Info 2021-10-22 13:31:19: Waiting for idle on (kernel): 340c1493-7f52-4f0a-92b0-e80c0c58ce7e -> idle
Info 2021-10-22 13:31:19: Finished waiting for idle on (kernel): 340c1493-7f52-4f0a-92b0-e80c0c58ce7e -> idle
Info 2021-10-22 13:31:19: UpdateWorkingDirectoryAndPath in Jupyter Notebook
Info 2021-10-22 13:31:19: changeDirectoryIfPossible
Info 2021-10-22 13:31:19: Adding hash for 1 = 94e31c57694e with 4 lines
Info 2021-10-22 13:31:19: Finished execution for e7dd6b67-4a2b-4677-84a7-4e4f83f385e3
Info 2021-10-22 13:31:28: UpdateWorkingDirectoryAndPath in Jupyter Notebook
Info 2021-10-22 13:31:28: Adding hash for 2 = dc925eb541a5 with 12 lines
Info 2021-10-22 13:31:28: Finished execution for 27648303-baef-463e-a9b7-ef8151adccf6
```

**Comment by DonJayamanne** [comment]
Should be fixed in the latest pre-release version
